include/nbl/builtin/hlsl/random/xoroshiro.hlsl

```diff
@@ -0,0 +1,47 @@
+
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_GLSL_RANDOM_XOROSHIRO_HLSL_INCLUDED_
+#define _NBL_BUILTIN_GLSL_RANDOM_XOROSHIRO_HLSL_INCLUDED_
+
+#include <nbl/builtin/hlsl/math/functions.hlsl>
+
+
+namespace nbl
+{
+namespace hlsl
+{
+
+#define xoroshiro64star_state_t uint2
+#define xoroshiro64starstar_state_t uint2
+
+void xoroshiro64_state_advance(inout uint2 state)
```

Comment:
all `inout` and `out` variable decls need to be replaced with `NBL_HLSL_REF(T)`

and we can use it in shared C++ as well
```cpp
#ifdef __cplusplus
#define NBL_HLSL_REF(T) std::add_reference_t<T>
#else
#define NBL_HLSL_REF(T) inout T
#endif
```

Reply:
@kpentaris works for you and the atomics?